### PR TITLE
Move lint check to separate stage

### PR DESCRIPTION
**Pull request by sxjscience**
*Description of changes:*

Move lint check to separate stage so the unit-test won't be triggered if the lint is failing.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by canerturkmen — APPROVED** [review]
looks good to me! fwiw, we may not want to parallelize the lint checks of submodules but have them in one box, likely because the overhead of spinning up a box is more than the check itself. this would hide away from the UI which package's lint is failing, but would save us resources (I know sometimes CI gets clogged because of lack of boxes).

I would also wait for someone familiar with CI before merging. Thanks!

**Comment by szha** [comment]
Job PR-1800-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1800/1/index.html

**Comment by yinweisu** [comment]
> looks good to me! fwiw, we may not want to parallelize the lint checks of submodules but have them in one box, likely because the overhead of spinning up a box is more than the check itself. this would hide away from the UI which package's lint is failing, but would save us resources (I know sometimes CI gets clogged because of lack of boxes).
> 
> I would also wait for someone familiar with CI before merging. Thanks!

Good point. For the current CI system we probably want to have them into one box. With the new CI, we should be able to parallel lint stages as well because we have "infinite" machines

File: Jenkinsfile
Comment (zhiqiangdon) [inline Jenkinsfile:68]: Is it better to change `mm` -> `multimodal`? We will use name `multimodal` for the new name space.

File: Jenkinsfile
Comment (sxjscience) [reply Jenkinsfile]: Fixed.

**Comment by szha** [comment]
Job PR-1800-2 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1800/2/index.html

**Review by yinweisu — APPROVED** [review]
LGTM

**Pull request merged**
